src/reducers/archiveThought.js

```diff
@@ -54,8 +56,18 @@ const archiveThought = (state, { path } = {}) => {
   if (!path) return state
 
   // same as in newThought
-  const contextChain = splitChain(state, path)
   const showContexts = isContextViewActive(state, contextOf(path))
+  if (showContexts) {
+    // Get thought in ContextView
+    const thoughtInContextView = head(contextOf(path))
+    // Get context from which we are going to delete thought
+    const context = getContexts(state, thoughtInContextView.value).map(({ context }) => context).find(context => head(context) === headValue(path))
+    if (context) {
+      // Convert to path
+      path = rankThoughtsFirstMatch(state, [...context, thoughtInContextView.value])
```

Comment:
In general, we can't convert a Context to a Path since the rank information has been lost. `rankThoughtsFirstMatch` infers the rank, choosing the first in a context if there are duplicates, which is not suitable for an exact path. `rankThoughtsFirstMatch` is used to infer the `cursor` from the URL and should not be used for any other purpose.

Reply:
My approach was to re-create Path from Context View to Path in Normal View. As i understand, we can't do it, because we lose rank information, so do i need to fix this issue in another way?

Reply:
I've tried this case:
- parentA
  - a
    - m
      - x
- parentB
  - b
    - m

In Context View we have [parentA, a] and [parentB, b].
We get path from cursor and in this case we don't have **parentB** there (the same problen as in https://github.com/cybersemics/em/pull/800). 
![Screenshot from 2020-07-15 18-03-25](https://user-images.githubusercontent.com/32064251/87537888-93474580-c6c5-11ea-842d-25d66bb4d17a.png)
We need either to restore path in Normal View from what we get or pass some additional data to cursor/reoragnize cursor for Context View in order to restore path from it. What do you think about it? If i'm kind of lost in path/cursor logic or forgot about some util functions that could help, please tell me :) Thanks!

Reply:
Gosh, this is difficult to talk about!

> We get path from cursor

The `cursor` *is* a `Path`. It is a transhierarchical `Path` in that it may cross multiple Context View boundaries.

> We get path from cursor and in this case we don't have **parentB** there (the same problen as in #800).

If I understand correctly, you're saying that when the `cursor` is `parentA/a/m~/b` then **parentB** is not explicitly stored in the `Path`, even though it is implicitly represented in the last simple (monohierarchical) `Path` of the `cursor`: `parentB/b/m`. Because it is not explicitly stored, we don't have all of the rank information.

> We need either to restore path in Normal View from what we get or pass some additional data to cursor/reoragnize cursor for Context View in order to restore path from it.

I'm a little lost at this point, but maybe this is because I lost the original context of this discussion.

The `contextChain` contains the full `Path` information for each segment in a transhierarchical `Path`. Thus, by deriving `contextChain` from `cursor`, you can recover the `Child` at any point in the `Path`.

Uh oh... `lastThoughtsFromContextChain` itself uses `rankThoughtsFirstMatch`. I wrote that function so I have nobody to blame but myself!

Reply:
@pushkov-fedor I can address the refactoring of `lastThoughtsFromContextChain` in a separate issue. We should still avoid all explicit uses of `rankThoughtsFirstMatch` here.